app/_kong_plugins/header-cert-auth/index.md

```diff
@@ -36,6 +36,144 @@ search_aliases:
+instead of relying on traditional TLS termination.
+This approach is particularly useful in scenarios where TLS traffic is terminated outside of {{site.base_gateway}} such as at an external CDN or load balancer and the client certificate is passed along in an HTTP header for subsequent validation.
+
+{:.warning}
+> **Important:** To use this plugin, you must add [certificate authority (CA) certificates](/gateway/entities/ca-certificate/). Set them up before configuring the plugin.
+
+## How it works
+
+This plugin addresses the inability of the {{site.base_gateway}} to authenticate API calls using client certificates received in HTTP headers, rather than through traditional TLS termination. 
+This occurs in scenarios where TLS traffic is not terminated at the {{site.base_gateway}}, but rather at an external CDN or load balancer, and the client certificate is preserved in an HTTP header for further validation.
+
+The Header Cert Authentication plugin is similar to the [mTLS Auth plugin](/plugins/mtls-auth).
+However, the mTLS plugin is only designed for traditional TLS termination, while the Header Cert Auth plugin also provides support for client certificates in headers. 
+
+The Header Cert Auth plugin extracts the client certificate from the HTTP header and validates it against the configured CA list. 
+If the certificate is valid, the plugin maps the certificate to a Consumer based on the common name field.
+
+The plugin validates the certificate provided against the configured CA list based on the
+requested Route or Gateway Service:
+* If the certificate is not trusted or has expired, the response is
+  `HTTP 401 TLS certificate failed verification`.
+* If a valid certificate is not presented (including when requests are not sent to the HTTPS port),
+  the response is `HTTP 401 No required TLS certificate was sent`.
+* However, if the `config.anonymous` option is configured on the plugin,
+  an [anonymous Consumer](/gateway/authentication/#using-multiple-authentication-methods) is used, and the request is allowed to proceed.
+
+The plugin can be configured to only accept certificates from trusted IP addresses, as specified by the [`trusted_ips`](/gateway/configuration/#trusted-ips) config option. This ensures that {{site.base_gateway}} can trust the header sent from the source and provides L4 level of security.
+
+{:.warning}
+> **Important:** Incomplete or improper configuration of the Header Cert Authentication plugin can compromise the security of your API.
+> <br><br>For instance, enabling the option to bypass origin verification can allow malicious actors to inject fake certificates, as {{site.base_gateway}} will not be able to verify the authenticity of the header. This can downgrade the security level of the plugin, making your API vulnerable to attacks. Ensure you carefully evaluate and configure the plugin according to your specific use case and security requirements.
```

Comment:
```suggestion
> <br><br>For instance, enabling the option to bypass origin verification can allow malicious actors to inject fake certificates, as {{site.base_gateway}} won't be able to verify the authenticity of the header. This can downgrade the security level of the plugin, making your upstream service vulnerable to attacks. Before using this plugin in production, carefully evaluate and configure the plugin according to your specific use case and security requirements.
```